bin/webmention

```diff
@@ -1,59 +1,68 @@
 #!/usr/bin/env ruby
 
-$:.unshift(File.join(File.dirname(__FILE__), "/../lib"))
-require "webmention"
-require "link_header"
+$LOAD_PATH.unshift(File.join(File.dirname(__FILE__), '..', 'lib'))
+
+require 'link_header'
+require 'webmention'
+
+def debug(msg)
+  puts msg
+end
 
 url = ARGV[0]
 
-if url == nil
-  puts "Usage: webmention http://example.com/post/100"
+if url.nil?
+  puts 'Usage: webmention https://source.example.com/post/100'
   exit 1
 end
 
-if !Webmention::Client.valid_http_url?(url)
-  puts "Not a valid URL"
+unless Webmention::Client.valid_http_url?(url)
+  puts "`#{url}` is not a valid URL."
   exit 1
 end
 
-client = Webmention::Client.new url
+client = Webmention::Client.new(url)
 
-def debug(msg)
-  puts msg
-end
+debug("Crawling `#{url}` for links...")
 
-debug "Finding links on #{url}"
 client.crawl
 
 links = client.links
-debug "Found #{links.length} links:"
-debug links.to_a.map{|m| "\t#{m}"}.join("\n")
 
-debug ""
+debug("Links found: #{links.length}")
+debug(links.to_a.map { |link| "\t#{link}" }.join("\n"))
+
+debug('')
 
 links.each do |link|
-  debug link
+  debug(link)
 
   endpoint = Webmention::Client.supports_webmention?(link)
-  if endpoint
-    debug "\tDiscovered Webmention endpoint:"
-    debug "\t#{endpoint}"
-    debug "\tSending webmention..."
-    result = Webmention::Client.send_mention endpoint, url, link, true
-    if result
-      debug "\tReturned HTTP #{result.code}"
-      if result.headers['link']
-        link_headers = LinkHeader.parse result.headers['link']
-        if status_link = link_headers.find_link(['rel', 'status'])
-          debug "\tWebmention status: #{status_link.href}"
-        end
-      end
-    else
-      debug "\tEncountered an unknown error sending this webmention!"
-    end
-  else
-    debug "\tNo webmention endpoint found"
+
+  unless endpoint
+    debug("\tNo webmention endpoint found at `#{link}`.")
+    next
+  end
+
+  debug("\tDiscovered webmention endpoint: #{endpoint}")
+  debug("\tSending webmention...")
+
+  result = Webmention::Client.send_mention(endpoint, url, link, true)
+
+  unless result
+    debug("\tEncountered an unknown error sending this webmention!")
+    next
+  end
+
+  debug("\tHTTP status code: #{result.code}")
+
+  link_headers = result.headers['link']
+
+  if link_headers
+    status_link = LinkHeader.parse(link_headers).find_link(%w[rel status])
+
+    debug("\tWebmention status URL: #{status_link.href}") if status_link
   end
 
-  debug ""
+  debug ''
 end
```

Comment:
No functionality changes in this file. The diff is gnarly, but it's mostly formatting and tidying up.